### ISSUE TITLE
Not clear that `fetch` is available in hooks

### ISSUE THREAD

**Issue by illright** [sveltejs/kit#2916]
### Describe the bug

The docs section on the hooks doesn't mention if `fetch` is available on the server side to use inside a hook (`handle()`, for example) and how to obtain a reference to it. A reference to _some_ `fetch()` is present in the sample code for the `externalFetch()` hook, but seeing as it's not a complete file, it leads to reader to believe this `fetch` might have to come from elsewhere, such as `node-fetch` installed additionally. 

I see several solutions to this issue:
1. Add a note to the documentation on hooks saying that `fetch` is globally available to use in any custom hook.
2. Rework the way the references to `fetch` are obtained. For example, create an internal module that you could import `fetch` from, so that `hooks.ts` would look like this:

   ```ts
   import { fetch } from '@sveltejs/kit/hooks';

   export function handle() {
     console.log(await fetch('https://kit.svelte.dev/'));
   }
   ```
   
I would prefer the second solution as it allows explicit reference to `fetch` as well as explicit typings for TypeScript. As a user of SvelteKit writing hooks, I would be more confident knowing that I can import `fetch` from somewhere to use in hooks instead of hoping there's a global one.

### Reproduction

There is no reproduction

### System Info

```shell
Irrelevant
```

### Severity

annoyance

**Comment by illright** [comment]
My story behind this bug is the following:

I was writing a `handle` hook that, in some cases, would make a request to the backend in order to authenticate a user. For that, I wanted to use `fetch`, but found no note of it in the documentation. VS Code was showing that there was a global `fetch` coming from `typescript/lib/lib.dom.d.ts`, so I assumed it was a mistake of TypeScript to believe that `hooks.ts` had access to DOM APIs. I tried installing `node-fetch`, but for some reason, the dev server wouldn't start up when `'node-fetch'` was imported due to peculiar `node:*` imports in its source code.

Frustrated, I cloned a fresh demo app, attempted to use a global fetch and it worked.
